x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/RoleMappingFileSettingsIT.java

```diff
@@ -163,23 +167,39 @@ public static void writeJSONFileWithoutVersionIncrement(String node, String json
         writeJSONFile(node, json, logger, versionCounter, false);
     }
 
-    private static void writeJSONFile(String node, String json, Logger logger, AtomicLong versionCounter, boolean incrementVersion)
+    public static void writeJSONFile(String node, String json, Logger logger, AtomicLong versionCounter, boolean incrementVersion)
         throws Exception {
         long version = incrementVersion ? versionCounter.incrementAndGet() : versionCounter.get();
 
         FileSettingsService fileSettingsService = internalCluster().getInstance(FileSettingsService.class, node);
-        assertTrue(fileSettingsService.watching());
-
-        Files.deleteIfExists(fileSettingsService.watchedFile());
 
         Files.createDirectories(fileSettingsService.watchedFileDir());
         Path tempFilePath = createTempFile();
 
         logger.info("--> before writing JSON config to node {} with path {}", node, tempFilePath);
         logger.info(Strings.format(json, version));
-        Files.write(tempFilePath, Strings.format(json, version).getBytes(StandardCharsets.UTF_8));
-        Files.move(tempFilePath, fileSettingsService.watchedFile(), StandardCopyOption.ATOMIC_MOVE);
-        logger.info("--> after writing JSON config to node {} with path {}", node, tempFilePath);
+
```

Comment:
Do you think it might be worth moving `writeJSONFile` to `ESIntegTestCase` and remove the duplicated function? (renaming it to something mode apt, like writeFileSettingsJson)?

Reply:
Yup that's a good call -- it's not too often that this method will be used but I think moving to `ESIntegTestCase` is still justified

Reply:
Discussed on Slack: this method is a little too specific for `ESIntegTestCase` so keeping it here after all